client/hotrod-client/src/main/java/org/infinispan/client/hotrod/iteration/SegmentKeyTracker.java

```diff
@@ -0,0 +1,55 @@
+package org.infinispan.client.hotrod.iteration;
+
+import org.infinispan.client.hotrod.impl.consistenthash.SegmentConsistentHash;
+import org.infinispan.client.hotrod.logging.Log;
+import org.infinispan.client.hotrod.logging.LogFactory;
+import org.infinispan.commons.util.Util;
+
+import java.nio.ByteBuffer;
+import java.util.BitSet;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.IntStream;
+
+import static java.nio.ByteBuffer.wrap;
+
+/**
+ * @author gustavonalle
+ * @since 8.0
+ */
+public class SegmentKeyTracker implements KeyTracker {
+
+   private static final Log log = LogFactory.getLog(SegmentKeyTracker.class);
+
+   private final ConcurrentMap<Integer, Set<ByteBuffer>> keysPerSegment = new ConcurrentHashMap<>();
```

Comment:
Given that lookups are far more frequent than removals, I will follow your suggestion, thanks @wburns